src/main/kotlin/racing/domain/Car.kt

```diff
@@ -1,12 +1,38 @@
 package racing.domain
 
 class Car(
-    val name: String
+    name: String
 ) {
-    var position: Int = 0
-        private set
+    val name = CarName(name)
+    val position = CarPosition()
 
     fun moveCar() {
-        position += 1
+        position.increase()
+    }
+}
+
+data class CarName(val name: String) {
+    init {
+        if (name.length > 5) {
+            throw IllegalArgumentException("자동차 이름은 5자를 초과할 수 없습니다.")
+        }
+    }
+}
+
+data class CarPosition(private var _position: Int = INIT_POSITION_NUMBER) : Comparable<CarPosition> {
+
+    val position: Int get() = _position
```

Comment:
private set을 위해 다음과 같이 내부 변수와 외부 공개용 프로퍼티를 따로 만들었습니다.

Reply:
```suggestion
    val position: Int
        get() = _position
```

위와 같이 작성하면 더 보기 편할 것 같아요~!